superset/db_engine_specs/base.py

```diff
@@ -1438,7 +1438,7 @@ def validate_parameters(
             return errors
 
         port = parameters.get("port", None)
-        if not port:
+        if not port or type(port) is str:
```

Comment:
What happens today if we pass a string as the port?

The proper way to handle validation would be adding to the Marshmallow schema, and we're already enforcing that it should be an int: https://github.com/apache/superset/blob/a241e318c6c1dfa94897858d7a2db4e7aa0247f3/superset/db_engine_specs/base.py#L1310

Reply:
currently, using the form, you can't pass in a string into the db-connection-modal because frontend side it is `type=number` so it only accepts numbers. 

I think the validation was if someone was using the api, and not the form. Would the marshmellow validation pick that up?

Reply:
You can test by either changing the frontend temporarily to allow passing a string, or by mocking the request.

To mock a request:

1. open the network tab
2. create a DB in the browser
3. right click the network request and click "copy as curl"
4. paste on a terminal, edit the command to pass a string as the port

Reply:
<img width="982" alt="Screen Shot 2021-06-14 at 1 43 26 PM" src="https://user-images.githubusercontent.com/48933336/121935585-994f0780-cd16-11eb-8849-ae93803cd355.png">